repository-module/src/main/java/com/gmail/zagurskaya/online/cash/repository/impl/UserRepositoryImpl.java

```diff
@@ -0,0 +1,152 @@
+package com.gmail.zagurskaya.online.cash.repository.impl;
+
+import com.gmail.zagurskaya.online.cash.repository.UserRepository;
+import com.gmail.zagurskaya.online.cash.repository.exception.UserRepositoryException;
+import com.gmail.zagurskaya.online.cash.repository.model.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class UserRepositoryImpl implements UserRepository {
+
+    private static Logger logger = LoggerFactory.getLogger(UserRepositoryImpl.class);
+
+    @Override
+    public User loadUserByUsername(Connection connection, String name) {
+        String sql = "SELECT * FROM users WHERE username = ?";
+        List<User> userList = new ArrayList<>();
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setString(1, name);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            return getUsersFromResult(resultSet).get(0);
```

Comment:
Need to use separate method got getting ONE result. You can get Indexofboundexception.